### PR TITLE
Better sanitize database queries in installer

### PR DESCRIPTION
**Pull request by fiammybe**

### CODE REVIEW

File: htdocs/install/page_dbsettings.php [hunk 1]
```diff
@@ -236,19 +236,19 @@ function select_db($db_name, $link) {
 	if (empty ($error)) {
 		if (!select_db($vars ['DB_NAME'], $link)) {
 			// Database not here: try to create it
-			$result = exec_query("CREATE DATABASE `" . $vars ['DB_NAME'] . '`', $link);
+			$result = exec_query("CREATE DATABASE `" . quote_sql($vars ['DB_NAME']) . '`', $link);
```
Comment (MekDrop) [inline htdocs/install/page_dbsettings.php:239]: quote_sql result already starts with quotes, so you need to remove them from string here

File: htdocs/install/page_dbsettings.php [hunk 2]
```diff
@@ -236,19 +236,19 @@ function select_db($db_name, $link) {
...
+				$error = sprintf(DATABASE_CREATED, quote_sql($vars ['DB_NAME']));
 				$db_exist = true;
 			}
 		} else {
 			$db_exist = true;
 		}
 		if ($db_exist && $vars['DB_CHARSET']) {
 			/* Attempt to set the character set and collation to the selected */
-			$sql = "ALTER DATABASE `" . $vars ['DB_NAME'] . "` DEFAULT CHARACTER SET " . quote_sql($vars ['DB_CHARSET']) . ($vars ['DB_COLLATION']?" COLLATE " . quote_sql($vars ['DB_COLLATION']):"");
+			$sql = "ALTER DATABASE `" . quote_sql($vars ['DB_NAME']) . "` DEFAULT CHARACTER SET " . quote_sql($vars ['DB_CHARSET']) . (quote_sql($vars ['DB_COLLATION'])?" COLLATE " . quote_sql($vars ['DB_COLLATION']):"");
```
Comment (MekDrop) [inline htdocs/install/page_dbsettings.php:251]: same (not ```"ALTER DATABASE `" . quote_sql($vars ['DB_NAME']) . "` DEFAULT CHARACTER SET "``` but ```"ALTER DATABASE " . quote_sql($vars ['DB_NAME']) . " DEFAULT CHARACTER SET "```)

**Comment by fiammybe** [comment]
This was found after investigation https://hackerone.com/reports/983710 by solov9ev

**Comment by fiammybe** [comment]
This was found after investingation https://hackerone.com/reports/983710 by solov9ev

**Review by MekDrop — CHANGES_REQUESTED** [review]
I found few small issues

File: htdocs/install/page_dbsettings.php (hunk 1 above)
Comment (fiammybe) [reply htdocs/install/page_dbsettings.php:239]: ok, didn't know it added quotes every time.

File: htdocs/install/page_dbsettings.php [hunk 3]
```diff
@@ -236,19 +236,19 @@ function select_db($db_name, $link) {
...
+				$error = sprintf(DATABASE_CREATED, quote_sql($vars ['DB_NAME']));
 				$db_exist = true;
 			}
 		} else {
 			$db_exist = true;
 		}
 		if ($db_exist && $vars['DB_CHARSET']) {
 			/* Attempt to set the character set and collation to the selected */
-			$sql = "ALTER DATABASE `" . $vars ['DB_NAME'] . "` DEFAULT CHARACTER SET " . quote_sql($vars ['DB_CHARSET']) . ($vars ['DB_COLLATION']?" COLLATE " . quote_sql($vars ['DB_COLLATION']):"");
+			$sql = "ALTER DATABASE " . quote_sql($vars ['DB_NAME']) . " DEFAULT CHARACTER SET " . $vars ['DB_CHARSET'] . ($vars ['DB_COLLATION']?" COLLATE " . $vars ['DB_COLLATION']:"");
```
Comment (MekDrop) [inline htdocs/install/page_dbsettings.php:251]: You have removed here a bit too much. You didn't needed to remove allmost all `quote_sql` only quotes.

File: htdocs/install/page_dbsettings.php (hunk 2 above)
Comment (fiammybe) [reply htdocs/install/page_dbsettings.php]: I changed the quote_sql with sanitize_database now

**Review by MekDrop — APPROVED** [review]

**Pull request merged**